src/app/modules/shared-components/components/path-suggestion/path-suggestion.component.ts

```diff
@@ -8,11 +8,18 @@ import { mapToFetchState } from '../../../../shared/operators/state';
 import { map } from 'rxjs/operators';
 import { itemRoute, urlArrayForItemRoute } from '../../../../shared/routing/item-route';
 import { UrlCommand } from '../../../../shared/helpers/url';
+import { typeCategoryOfItem } from '../../../../shared/helpers/item-type';
 
 const getItemRouteUrl = (id: string, breadcrumbs: BreadcrumbsFromRoot[]): UrlCommand => {
   const index = breadcrumbs.findIndex(item => item.id === id);
   const path = breadcrumbs.slice(0, index).map(item => item.id);
-  return urlArrayForItemRoute(itemRoute('activity', id, path));
+  const firstItem = breadcrumbs[0];
+
+  if (!firstItem) {
+    throw new Error('Unexpected: Missed breadcrumbs from root');
+  }
```

Comment:
I don't get this check.  
What are you trying to avoid ?  Empty `breadcrumbs` array? Then just do `if (!breadcrumbs.length) throw ...` on line 14.  

EDIT: ah, got it, you use it on line 22 and you do not want it to be possibly `undefined`...
but why the first element of the breadcrumb? You need the type of the element you are linking to and that's not the one of the first element of the breadcrumb!

Reply:
Fixed. Also I've add one more case on "Prerequisites" page